pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -109,12 +109,16 @@ func (h *HeapsterMetricsClient) GetCPUUtilization(namespace string, selector map
 	return &utilization, timestamp, nil
 }
 
-func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace string, selector map[string]string) (avgConsumption int64,
+func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace string, selector string) (avgConsumption int64,
 	avgRequest int64, timestamp time.Time, err error) {
 
-	labelSelector := labels.SelectorFromSet(labels.Set(selector))
+	// TODO: Eliminate deserialization/reserialization of the selector.
```

Comment:
Yes, I'm aware of that. I want to pass it unparsed, but would rather not block 1.2 on that.